test/test_jit.py

```diff
@@ -16092,6 +16092,14 @@ def create_nn_module(*args, **kwargs):
         args_variable, kwargs_variable = create_input(input, dtype=dtype)
         f_args_variable = deepcopy(unpack_variables(args_variable))
 
+        # TODO(issue####) Neither this nor no_grad should be required
+        # if check_against_reference() is updated to check gradients
+        # w.r.t. weights and then only check w.r.t. inputs if any
+        # inputs require it.
+        nonlocal no_grad
```

Comment:
why do you need the nonlocal here? You want to update this only for the current function right?

Reply:
Does it make a difference? The function is only created once per top-level call. Either way, it's sloppy and potentially risky. Does this look better?

Reply:
Yes much better thanks!